### PR TITLE
Issue 321: Fix syslog messages get merged

### PR DESCRIPTION
**Pull request by AnDann**
Try to fix issue 321
debug stream seems to work, but other functions are still WIP.

### CODE REVIEW

File: protocols/syslog/syslog.c [hunk 1]
```diff
@@ -34,88 +35,142 @@
 #include "syslog_net.h"
 
 
-static char send_buffer[MAX_DYNAMIC_SYSLOG_BUFFER + 1];
+static char send_buffer[MAX_LINES_SYSLOG_BUFFER][MAX_LENGTH_SYSLOG_BUFFER + 1];
```
Comment (eku) [inline protocols/syslog/syslog.c]: I suggest to not hard-code the number of buffers and thus waste memory. Lets dynamically allocate and free it and define a maximum number of allocated buffers.

File: protocols/syslog/syslog.c [hunk 2]
```diff
@@ -34,88 +35,142 @@
...
 extern uip_udp_conn_t *syslog_conn;
 static struct SyslogCallbackCtx syslog_callbacks[SYSLOG_CALLBACKS];
```
Comment (eku) [inline protocols/syslog/syslog.c]: Does it make sense to hold the send_buffer and the callback pointer in a structure, that is dynamicall allocated? You still have to remember the allocated structures. I suggest a single linked list with a global variable that points to the first entry in the list.

File: protocols/syslog/syslog.c [hunk 3]
```diff
@@ -34,88 +35,142 @@
...
+int8_t act_char_line = -1;
 
-static void syslog_send_cb_P(void *data) 
+static void
+syslog_send_cb_P(void *data)
 {
   strcpy_P(uip_appdata, data);
```
Comment (eku) [inline protocols/syslog/syslog.c]: What happens if the UIP buffer is smaller than the string? Should we better user strncpy?

**Comment by eku** [comment]
Looks like rewriting the code is the best solution rather working on symptoms.

**Comment by AnDann** [comment]
How about sending out the received data direct instead of storing it in a buffer. This would save the buffers and the callback structures.

```c
uint8_t
syslog_send(const char *message)
{
#ifdef ETHERNET_SUPPORT
  if (!syslog_conn || uip_check_cache(&syslog_conn->ripaddr))
    return;                     /* ARP cache not ready, don't send request
                                 * here (would flood, wait for poll event). */
#endif /* ETHERNET_SUPPORT */

  uip_slen = 0;
  uip_appdata = uip_sappdata = uip_buf + UIP_IPUDPH_LEN + UIP_LLH_LEN;

  strcpy(uip_appdata, message);
  uip_udp_send(strlen(message));

  if (!uip_slen)
    return;

  uip_udp_conn = syslog_conn;
  uip_process(UIP_UDP_SEND_CONN);
  router_output();

  uip_slen = 0;
}
```

**Comment by eku** [comment]
Since syslog sends via UDP your proposed solution would work. I wonder why the author of the module chosed the buffered solution. Anyway, we should test your solution intensively.

BTW, using strcpy is dangereous. Always keep in mind issue #248.

**Comment by AnDann** [comment]
What happens when debug_printf in uip or while sending or receiving from other applications is called.

**Pull request closed without merging**